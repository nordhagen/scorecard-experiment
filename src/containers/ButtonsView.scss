@import '/global.scss';

.ButtonsView {
  width: 70vw;
  height: 100vh;
  padding: 5px;
}
