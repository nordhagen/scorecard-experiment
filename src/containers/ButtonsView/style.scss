@import '/global.scss';

.ButtonsView {
  width: 70vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
}

.kahoot {
  font-weight: 600;
}

.buttons {
  padding: 5px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-grow: 1;
}
