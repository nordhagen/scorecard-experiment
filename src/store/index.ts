import { createStore, combineReducers } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import { scoreConfigReducer } from './config/reducers'
import { scoreStateReducer } from './score/reducers'

const rootReducer = combineReducers({
  score: scoreStateReducer,
  config: scoreConfigReducer
})

export type AppState = ReturnType<typeof rootReducer>
const store = createStore(rootReducer, composeWithDevTools())

export default store
