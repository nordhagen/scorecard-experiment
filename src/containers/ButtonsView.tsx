import * as React from 'react'
import styles from './ButtonsView.scss'
import ScoreButton from '../components/ScoreButton'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { AppState } from '../store'
import { addScoreItem } from '../store/score/actions'
import { ScoreConfig } from '../store/config/types'

interface ButtonsViewProps {
  config: ScoreConfig
  addScoreItem: typeof addScoreItem
}

class ButtonsView extends React.Component<ButtonsViewProps> {
  colorMap = {
    a: '#ff0000',
    b: '#00ff00',
    c: '#0000ff',
    d: '#ffff00'
  }

  render() {
    const buttons = this.props.config.items.map(item => (
      <ScoreButton
        id={item.type}
        key={item.type}
        clickDelegate={this.props.addScoreItem}
        color={this.colorMap[item.type]}
      />
    ))

    return (
      <section className={styles.ButtonsView}>
        <header>
          <h1>Kahoot! Points</h1>
        </header>
        <div className={styles.buttons}>{buttons}</div>
      </section>
    )
  }
}

const mapStateToProps = (state: AppState) => ({
  config: state.config
})

const mapDispatchToProps = dispatch =>
  bindActionCreators({ addScoreItem }, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ButtonsView)
