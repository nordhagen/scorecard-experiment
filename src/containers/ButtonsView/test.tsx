import React from 'react'
import ButtonsView from '.'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import store from '../../store'

test('It renders correctly', () => {
  const pastItem = renderer.create(
    <Provider store={store}>
      <ButtonsView />)
    </Provider>
  )
  expect(pastItem.toJSON()).toMatchSnapshot()
})
