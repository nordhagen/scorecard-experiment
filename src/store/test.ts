import store from '.'

test('initial state', () => {
  let state
  state = store.getState()
  expect(state).toEqual({
    score: {
      items: [
        { type: 'a', count: 0 },
        { type: 'b', count: 0 },
        { type: 'c', count: 0 },
        { type: 'd', count: 0 }
      ]
    },
    config: {
      items: [
        { type: 'a', unitPoints: 50, bonusTrigger: 3, bonusValue: 200 },
        { type: 'b', unitPoints: 30, bonusTrigger: 2, bonusValue: 90 },
        { type: 'c', unitPoints: 20, bonusTrigger: null, bonusValue: null },
        { type: 'd', unitPoints: 15, bonusTrigger: null, bonusValue: null }
      ]
    }
  })
})
