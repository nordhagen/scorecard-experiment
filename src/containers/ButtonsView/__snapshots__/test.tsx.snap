// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
Array [
  <section>
    <header>
      <h1>
        <span>
          Kahoot!
        </span>
         Points
      </h1>
    </header>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F27781",
          }
        }
      >
        a
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#04BF8A",
          }
        }
      >
        b
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F2CF1D",
          }
        }
      >
        c
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F29F05",
          }
        }
      >
        d
      </button>
    </div>
  </section>,
  ")",
]
`;
